vernac/declare.ml

```diff
@@ -90,8 +91,17 @@ module Info = struct
      start of the proof in the interactive case. *)
   let make ?(poly=false) ?(inline=false) ?(kind=Decls.(IsDefinition Definition))
       ?(udecl=UState.default_univ_decl) ?(scope=Locality.default_scope)
+      ?(clearbody=false)
       ?hook ?typing_flags () =
-    { poly; inline; kind; udecl; scope; hook; typing_flags }
+    let () = match scope, clearbody with
+      | Discharge, true ->
```

Comment:
Can't we just put clearbody in the `Discharge` constructor? IMHO that's better it doesn't make sense for Global.

Reply:
We could put it in the Discharge, although that type is also used for variables (ie no body) so I wasn't sure it would make much sense.
OTOH passing `clearbody` alongside it isn't better for that

Reply:
Yeah maybe it has to go in the `SectionLocalDef` constructor? Would be good if we can find a way to avoid the assert etc... in the end the `clearbody` flag is used in a very very specific place.

Reply:
I put it in SectionLocalDef, although it's still also in Info.t as SectionLocalDef is created late.
I moved this check to the attribute handling (synterp.ml defattributes)